Src/C#-SIMS/Inventory.cs

```diff
@@ -0,0 +1,217 @@
+        Console.WriteLine("****View all products****");
+
+        int index = 1;
+        foreach (var product in products)
+        {
+            Console.Write($"{index++}: ");
+            Console.WriteLine(product.ViewProductDetail());
+        }
+        ReturnToMainMenu();
+    }
+
+    private static void EditProduct()
+    {
+        initUI();
+        string? nameIO;
+        Console.WriteLine("****Edit a product****");
+        int index = 1;
+        foreach (var product in products)
+        {
+            Console.Write($"{index++}: ");
+            Console.WriteLine(product.Name);
+        }
+        Console.Write("PLease write the product's name: ");
+        nameIO = Console.ReadLine();
+        Product? productToEdit = products.Where(p => p.Name == nameIO).FirstOrDefault();
+        if (productToEdit != null)
+        {
+            Console.WriteLine("Please choose the number of what you want to change. (You can choose multiple choices, just separate them with commas.):");
+            Console.Write("1: Name\n2: Price\n3: Quantity\n: ");
+            string? optionsIO = Console.ReadLine();
+            List<string> itemsList = new List<string>(optionsIO?.Split(',') ?? Array.Empty<string>());
+            do
+            {
+                switch (itemsList[0])
+                {
+                    case "1":
+                        Console.Write("Please enter the new name: ");
+                        string? name = Console.ReadLine();
+                        productToEdit.Name = name;
+                        break;
+                    case "2":
+                        Console.Write("Please enter the new price: ");
+                        int price = int.Parse(Console.ReadLine() ?? "0");
+                        productToEdit.Price = price;
+                        break;
+                    case "3":
+                        Console.Write("Please enter the new quantity: ");
+                        int quantity = int.Parse(Console.ReadLine() ?? "0");
+                        productToEdit.Quantity = quantity;
+                        break;
+                }
+                itemsList.RemoveAt(0);
+            } while (itemsList.Count != 0);
+            Console.WriteLine("The product changed successfully.....\n");
+        }
+        else
+        {
+            Console.WriteLine("The product could bot be found please check the name and try again.");
+        }
+        ReturnToMainMenu();
+
+    }
+    private static void DeleteProduct()
+    {
+        initUI();
+        string? nameIO;
+        Console.WriteLine("****Delete a product****");
+        int index = 1;
+        foreach (var product in products)
+        {
+            Console.Write($"{index++}: ");
+            Console.WriteLine(product.Name);
+        }
+        Console.Write("PLease write the product's name: ");
+        nameIO = Console.ReadLine();
+        Product? productToDelete = products.Where(p => p.Name == nameIO).FirstOrDefault();
```

Comment:
You're totally right! Indexing is all I need. I messed up because I thought I had to create the object first and then put it in the list. But no, I'm creating and adding it to the list at the same time. So, just removing it from the list will do the trick. No object left behind